fdbserver/storageserver.actor.cpp

```diff
@@ -1521,7 +1521,14 @@ Future<Version> waitForVersion(StorageServer* data, Version commitVersion, Versi
 		return transaction_too_old();
 	} else {
 		if (commitVersion < data->oldestVersion.get()) {
-			return data->oldestVersion.get();
+			if (data->version.get() < readVersion) {
+				// Majority of the case, try using higher version to avoid
+				// transaction_too_old error when oldestVersion advances.
+				return data->version.get(); // majority of cases
```

Comment:
I like this solution. This is correct because this storage server doesn't have any newer commits in the range (commitVersion,  readVersion] and so can read at any version in this range.